shell/mixins/child-hook.js

```diff
@@ -24,15 +24,43 @@ export default {
       this._registerHook(AFTER_SAVE_HOOKS, boundFn, name, priority, boundFnContext);
     },
 
+    /**
+     * Applies hooks for a given key, omitting provided hooks
+     * @param key Key that specifies the kind of hook, e.g. _beforeSaveHooks,
+     * _afterSaveHooks
+     * @param args Arguments supplied to the `applyHooks` function. Hooks can be
+     * omitted by supplying an `omitHooks` object as the final argument.
+     * `omitHooks` describes which hooks need to be omitted for the provided
+     * key. Contains two properties:
+     *   1. _omitHooksMarker: boolean - Signals that this object contains
+     *      information about hooks that need to be omitted
+     *   2. names: string[] - A collection of names for hooks that should be
+     *      omitted
+     */
     async applyHooks(key, ...args) {
-      if ( !key ) {
+      let omitHooks = [];
+
+      // Check if the last argument is an object with the _omitHooksMarker property
+      if (
+        args.length > 0 &&
```

Comment:
args are optional, so think this should be args?.length.

though in this context is it a `parameter` kind of thing which is never null?

Reply:
Correct, `args?.length` will suffice here. Fixed.